app.rb

```diff
@@ -27,8 +27,9 @@
 						 :email => params[:email],
 						 :password => params[:password] )
 	if user.save
-    session[:id] = user.id
-		redirect "/profile/#{user.username}"
+    	session[:id] = user.id
+    	session[:username] = user.username
+		redirect '/profile'
```

Comment:
oh oops i was asleep and in bed when i wrote the above comment! you're right, ill fix that now